### DISCUSSION TITLE
Do you consider supporting local imports?

### DISCUSSION THREAD

**Discussion by pwtail** [gleam-lang/gleam#2072] [Ideas & suggestions]
I mean, to be able import a name within a function only.

**Comment by lpil** [comment]
Possibly! What is your use case?

**Comment by pwtail** [comment]
I think I have a very nice idea on how we can tame Ecto. I mean, writing an adapter for gleam.

Consider the first example here, movies and characters https://hexdocs.pm/ecto/associations.html

Every model (schema) can be associated with a very simple module (gleam or bare Erlang):

```
// movie.gleam

pub fn title() {
  #(field("string"), primary_key(True))
}

pub fn release_date() {
  #(field("date"))
}

pub fn characters() {
  #(has_many("character"))
}

```

This can be auto-generated from the Ecto schema, or inverse, be the source for the generated Ecto schema.

Now, how do we do queries:

```
import models/movie as m
import models/character as c
query = #(
  c.associate_to(m.characters),
  join(m.id, c.movie),   // I think Ecto can figure that out by itself
  gt(m.release_date, start_date),
  ilike(m.title, "%summer%"),
)
|> m.query
```

What do you think?

**Reply by lpil** [reply]
Why is it that you could not import at the top level in this case with Ecto?

**Reply by pwtail** [reply]
table aliases are usually just one letter and can coincide for multiple models (person and post, for example). And they just pollute the namespace

**Reply by lpil** [reply]
We discourage single letter variable names, so if this feature is to enable them then it would be best if we do not add them.

**Reply by pwtail** [reply]
Do you have an alternative idea on how Ecto can be adapted?

**Reply by lpil** [reply]
I don't understand how the code example given relates to Ecto or local imports I'm afraid.

**Reply by pwtail** [reply]
An Ecto query is a map. I composed a tuple `#(...)` that can be transformed to it. `m.id`, `m.release_date`, `m.characters` are all functions. You can call them and get the necessary info about a given field. And one-letter variables are a critical necessity. For a good user interface

**Reply by lpil** [reply]
I'm sorry, I don't understand. Why would the code break without a single letter variable name? Have you written a parser for Gleam and require single letters in this context for some reason?

**Reply by pwtail** [reply]
That harms the user interface, period (not being able to make short variables).

**Reply by lpil** [reply]
Single letter names are discouraged, but if you want them you can create them with `import my/module as m`
